openrtb/macrosubs.go

```diff
@@ -96,47 +70,18 @@ var findMatchesPool = sync.Pool{
 // It takes a string which the substitutions should be performed on, and a *BidResponse to determine the values to be substituted.
 // MacroSubs assumes that the BidResponse has exactly one seat, which has exactly one bid.
 // If this is not true, it will return empty string and an error.
-func MacroSubs(stringToSub string, bidRes *BidResponse, s Settlement) (result string, err error) {
-	subValues, err := createSubValues(bidRes, s)
-	if err != nil {
-		return "", err
+func MacroSubs(stringToSub string, response *BidResponse, seat *SeatBid, bid *Bid, settlement Settlement) string {
+	m := map[macro]string{
+		auctionID:       settlement.AuctionID(),
+		auctionBidID:    bid.ID,
+		auctionImpID:    bid.ImpressionID,
+		auctionSeatID:   seat.Seat,
+		auctionAdID:     bid.AdID,
+		auctionPrice:    strconv.FormatFloat(settlement.Price(), 'f', 9, 64),
+		auctionCurrency: string(settlement.Currency()),
 	}
-
-	replacer := createReplacer(subValues)
-	var re = findMatchesPool.Get().(*regexp.Regexp)
-	defer findMatchesPool.Put(re)
-	result = re.ReplaceAllStringFunc(stringToSub, replacer)
-
-	return
-}
-
-// CreateMacroSubstituter instantiates a MacroSubstituter object from a BidResponse.
-// Use this when you are going to be doing many substitutions using the same BidResponse data.
-// CreateMacroSubstituter assumes that the BidResponse has exactly one seat, which has exactly one
-// bid. If this is not true, it will return nil and an error.
-func CreateMacroSubstituter(bidRes *BidResponse, s Settlement) (substituter *MacroSubstituter, err error) {
-	subValues, err := createSubValues(bidRes, s)
-	if err != nil {
-		return nil, err
-	}
-
-	return &MacroSubstituter{replacer: createReplacer(subValues)}, nil
-}
-
-// SubstituteSlice applies macro substitution on a slice of strings. The strings in the slice are mutated;
-// this method does not return anything.
-func (substituter *MacroSubstituter) SubstituteSlice(strings []string) {
-	var re = findMatchesPool.Get().(*regexp.Regexp)
-	defer findMatchesPool.Put(re)
-	for i := range strings {
-		strings[i] = re.ReplaceAllStringFunc(strings[i], substituter.replacer)
-	}
-}
-
-// SubstituteString applies macro substitution on a string reference. Unlike the MacroSubs
-// function, this mutates the string and does not return anything.
-func (substituter *MacroSubstituter) SubstituteString(str *string) {
+	replacer := createReplacer(m)
 	var re = findMatchesPool.Get().(*regexp.Regexp)
 	defer findMatchesPool.Put(re)
-	*str = re.ReplaceAllStringFunc(*str, substituter.replacer)
+	return re.ReplaceAllStringFunc(stringToSub, replacer)
```

Comment:
How was the benchmarking for not having pooled replacers? Any remarks? I rather have simpler code indeed unless the change is significant.